sdk_container/src/third_party/coreos-overlay/eclass/coreos-kernel.eclass

```diff
@@ -169,19 +192,20 @@ setup_keys() {
 	EOF
 	openssl req -new -nodes -utf8 -days 36500 -batch -x509 \
 		"-${sig_hash}" -outform PEM \
-		-config certs/modules.cnf \
-		-out certs/modules.pub.pem \
-		-keyout certs/modules.key.pem \
+		-config gen_certs/modules.cnf \
+		-out gen_certs/modules.pub.pem \
+		-keyout gen_certs/modules.key.pem \
 		|| die "Generating module signing key failed"
-	cat certs/modules.pub.pem certs/modules.key.pem > "${sig_key}"
-}
 
-# Discard the module signing key but keep public certificate.
-shred_keys() {
-	local sig_key
-	sig_key="build/$(getconfig MODULE_SIG_KEY)"
-	shred -u certs/modules.key.pem "${sig_key}" || die
-	cp certs/modules.pub.pem "${sig_key}" || die
+	# copy the cert/key to desired location
+	mkdir -p "${MODULES_SIGN_CERT%/*}" "${MODULES_SIGN_KEY%/*}" || die
+	cat gen_certs/modules.pub.pem gen_certs/modules.key.pem > "$MODULES_SIGN_KEY" || die
```

Comment:
why is the public key prepended to the private key?

Reply:
I've just taken the original logic from the existing eclass: https://github.com/flatcar/scripts/blob/32416aabbcb756c6bdd2231dd2e39dac7097978b/sdk_container/src/third_party/coreos-overlay/eclass/coreos-kernel.eclass#L176
Gentoo also seems to do the same thing:
https://github.com/gentoo/gentoo/blob/b64e85fb80f47f5a12ca90947069eb433d212540/eclass/kernel-build.eclass#L166